helixbass/tree-sitter-lint
# Provided types first pass

In this PR:
- get "provided types" apparently working

Not in this PR:
- code generation for custom local binary for provided types stuff

To test:
The custom local project has a silly rule committed with some debug-logging so if you run `cargo build --release` from `.tree-sitter-lint/tree-sitter-lint-local/` and then run `cargo run` from the repo root you should see that it was able to successfully instantiate a "provided type" (and only did so once per file) that that rule leverages to print out violations (one per encountered `use`, it makes no sense) for any files that start with `use `

Based on `get-first-token`
